cmd/juno/juno.go

```diff
@@ -248,7 +248,7 @@ func NewCmd(config *node.Config, run func(*cobra.Command, []string) error) *cobr
 		v.SetEnvPrefix("JUNO")
 		v.SetEnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_"))
 		if err := v.BindPFlags(cmd.Flags()); err != nil {
-			return nil
+			return err
```

Comment:
is it expected?

Reply:
I traced what [v.BindPFlags](https://github.com/spf13/viper/blob/a0ece6a20fd5509209614eb8fc2d1f2cd212ec61/viper.go#L1049) does and found that its return ultimately depends on what [v.BindFlagValues](https://github.com/spf13/viper/blob/a0ece6a20fd5509209614eb8fc2d1f2cd212ec61/viper.go#L1071) returns, which is `nil`. So I think there is no difference between returning `nil` or `err`. Please double check me.

The `nilerr` linter discourages returning `nil` when `err` is not `nil`, so alternatively, I can revert this change and add a `// nolint:nilerr`

Reply:
I'd add a `// nolint:nilerr` for now and decide what to do later because it's safer, but that's me since I don't have much experience with the project.

@kirugan thoughts?

Reply:
I agree, let's revert this and add nolint comment

Reply:
Btw I just checked impl in Viper and seems like they have a but in there. TL;DR: they always return error=nil for this function